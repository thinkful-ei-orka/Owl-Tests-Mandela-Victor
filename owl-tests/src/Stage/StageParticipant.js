import React from 'react';
import participants from '../store'

export default function StageParticipant(props) {
     return (
          <div className='Stage-Participant' id={props.id}>
               <h3>
                    {props.name}
               </h3>
               <img src={props.avatar}></img>
          </div>
     )
}