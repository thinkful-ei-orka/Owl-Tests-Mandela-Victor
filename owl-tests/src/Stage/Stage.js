import React from 'react';
import StageParticipant from './StageParticipant';
import store from '../store'


import './Stage.css';

export default function Stage(props) {
    const participants = store.participants.map(participant => {
        if (participant.onStage) {
            return (
                <StageParticipant id={participant.id} avatar={participant.avatar} name={participant.name}/>
            )
        } else 
        return null
    })

         return (
             <div className='Stage' key={props.id}>
                <div className='Stage-Participants'>
                    {participants}
                </div>
             </div>
         );
     }