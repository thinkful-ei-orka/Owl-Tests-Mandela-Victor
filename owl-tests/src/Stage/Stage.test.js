import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import Stage from '../Stage/Stage'

it('renders', () => {
     const div = document.createElement('div');
     ReactDOM.render(<Stage />, div);
     ReactDOM.unmountComponentAtNode(div);
})

it('renders', () => {
     const tree = renderer
     .create(<Stage className='Stage'/>)
     .toJSON();
     expect(tree).toMatchSnapshot();
})