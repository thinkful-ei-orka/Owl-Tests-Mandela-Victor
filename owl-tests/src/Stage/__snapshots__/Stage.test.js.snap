// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
<div
  className="Stage"
>
  <div
    className="Stage-Participants"
  >
    <div
      className="Stage-Participant"
      id={1}
    >
      <h3>
        Koren Templeton
      </h3>
      <img
        src="https://robohash.org/itaquevoluptatumrerum.jpg?size=200x200&set=set1"
      />
    </div>
    <div
      className="Stage-Participant"
      id={5}
    >
      <h3>
        Ashla Attwool
      </h3>
      <img
        src="https://robohash.org/iustodoloremqueinventore.jpg?size=200x200&set=set1"
      />
    </div>
  </div>
</div>
`;
